baseplate/core.py

```diff
@@ -317,19 +324,59 @@ def __exit__(self, exc_type, value, traceback):
         else:
             self.finish()
 
+    def make_child(self, name, local=False, component_name=None):
+        """Return a child Span whose parent is this Span.
 
-class ServerSpan(Span):
-    """A server span represents a request this server is handling.
+        The child span can either be a local span representing an in-request operation
+        or a span representing an outbound service call.
 
-    The server span is available on the :term:`context object` during requests
-    as the ``trace`` attribute.
+        In a server, a local span represents the time spent within a
+        local component performing an operation or set of operations.
+        The local component is some grouping of business logic,
+        which is then split up into operations which could each be wrapped
+        in local spans.
 
-    """
+        """
+        raise NotImplementedError
 
-    def make_child(self, name):
-        """Return a child span representing an outbound service call."""
+
+class LocalSpan(Span):
+    # pylint: disable=invalid-name
+    def __init__(self,
+                 trace_id,
+                 parent_id,
+                 span_id,
+                 sampled,
+                 flags,
+                 name,
+                 context):
+        super(LocalSpan, self).__init__(trace_id, parent_id, span_id, sampled,
+                                        flags, name, context)
+        self.context = context
+
+    def make_child(self, name, local=False, component_name=None):
         span_id = random.getrandbits(64)
-        span = Span(self.trace_id, self.id, span_id, self.sampled, self.flags, name)
+
+        if local:
+            context_copy = self.context.clone()
+            span = LocalSpan(self.trace_id, self.id, span_id, self.sampled,
+                             self.flags, name, context_copy)
+            if component_name is None:
+                raise SpanError("Cannot create local span without component name.")
```

Comment:
Good call on `ValueError`. Those docs got accidentally removed while shuffling things around, adding back in. To answer here, the `component_name` is a secondary namespace of the local span. The span name itself should be the specific operation, but you might want to query similar operations under different namespaces ("components"), e.g. if you had "query" named spans for both "cassandra" and "mysql" components. http://javadox.com/io.zipkin.java/zipkin/1.1.4/zipkin/Constants.html#LOCAL_COMPONENT